### PR TITLE
WIP: feat: add installation on Windows

### LINKED ISSUE #12: dataLogDir is set to same directory as log4j
dataLogDir is for transaction logs. In this role, it's value is set to `zk_log_dir`, the same directory assigned for logs by log4j.

There should be a separate `zk_datalog_dir` variable for the dataLogDir property.

https://zookeeper.apache.org/doc/r3.1.2/zookeeperAdmin.html#sc_advancedConfiguration

### PR DESCRIPTION
**Pull request by koushyk**
Add installation on Windows #10 
Fixed #12

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -16,16 +16,20 @@ zk_autopurge_snapRetainCount: 10
...
-zk_tarball_dir: /opt/src
+zk_tarball_dir: '{{ zk_default_tarball_dir }}'
 zk_rolling_log_file_max_size: 10MB
 zk_max_rolling_log_file_count: 10
 zk_inventory_group: zookeeper
 
 zk_service_name: zookeeper
 zk_service_start: true
 zk_service_autostart: true
+chocolatey_installer: https://chocolatey.org/install.ps1
```
Comment (pavelpikta) [inline defaults/main.yml]: Unused variable

File: defaults/main.yml [hunk 2]
```diff
@@ -16,16 +16,20 @@ zk_autopurge_snapRetainCount: 10
...
+chocolatey_path: c:/ProgramData/chocolatey
```
Comment (pavelpikta) [inline defaults/main.yml]: Unused variable

File: defaults/main.yml [hunk 3]
```diff
@@ -16,16 +16,20 @@ zk_autopurge_snapRetainCount: 10
...
+chocolatey_version: latest
```
Comment (pavelpikta) [inline defaults/main.yml]: Unused variable

File: defaults/main.yml [hunk 4]
```diff
@@ -16,16 +16,20 @@ zk_autopurge_snapRetainCount: 10
...
+chocolatey_windows_compression: "false"
```
Comment (pavelpikta) [inline defaults/main.yml]: Unused variable

File: molecule/cloud-aws-windows/molecule.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      - windows
+provisioner:
+  name: ansible
+  log: false
+  config_options:
+    defaults:
+      callback_whitelist: profile_tasks,timer
+  connection_options:
+    ansible_connection: winrm
+  inventory:
```
Comment (pavelpikta) [inline molecule/cloud-aws-windows/molecule.yml]: add test for tarball installation

File: tasks/Win32NT/system.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
+---
+- name: Debug
```
Comment (pavelpikta) [inline tasks/Win32NT/system.yml]: Please remove

File: tasks/Win32NT/tarball_install.yml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  win_file:
+    path: '{{ zk_dir }}'
+    state: directory
+
+- name: "Install nssm"
+  win_chocolatey:
+    name: nssm
+    state: present
+
+- name: Install PSCX
```
Comment (pavelpikta) [inline tasks/Win32NT/tarball_install.yml]: Duplicated code

File: templates/Win32NT/zoo.cfg.j2 [hunk 8]
```diff
@@ -0,0 +1,28 @@
+# The number of milliseconds of each tick
```
Comment (pavelpikta) [inline templates/Win32NT/zoo.cfg.j2]: Can  use one template for windows and linux?

File: defaults/main.yml [hunk 9]
```diff
@@ -16,12 +16,12 @@ zk_autopurge_snapRetainCount: 10
...
-zk_data_dir: /var/lib/zookeeper
-zk_log_dir: /var/log/zookeeper
-zk_dir: '{{ zk_tarball_installation | ternary("/opt/zookeeper-" + zk_version, "/usr/lib/zookeeper") }}'
-zk_force_myid: true
-zk_force_config: true
-zk_tarball_dir: /opt/src
+zk_data_dir: '{{ zk_default_data_dir }}'
+zk_log_dir: '{{ zk_default_log_dir }}'
+zk_dir: '{{ zk_default_dir }}'
+zk_force_myid: false
```
Comment (pavelpikta) [inline defaults/main.yml]: why ?

File: molecule/cloud-aws-windows-choco/molecule.yml [hunk 10]
```diff
@@ -0,0 +1,71 @@
+---
```
Comment (pavelpikta) [inline molecule/cloud-aws-windows-choco/molecule.yml]: combine test in one molecule scenario

File: templates/zoo.cfg.j2 [hunk 11]
```diff
@@ -26,7 +26,7 @@ clientPort={{ zk_client_port }}
 {% if (zk_version is version_compare('3.5.0', '<')) %}
 {%- set ips = [] %}
 {%- for host in groups[zk_inventory_group] %}
-{{- ips.append(dict(id=loop.index, host=host, ip=hostvars[host]['ansible_default_ipv4'].address)) }}
+{{- ips.append(dict(id=loop.index, host=host, ip=hostvars[host]['ansible_fqdn'])) }}
```
Comment (ViachaslauKabak) [inline templates/zoo.cfg.j2]: why ip was changed to fqdn?

File: vars/Win32NT.yml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+---
+zk_install_type: '{{ zk_tarball_installation | ternary("tarball_install", "choco_install") }}'
+zk_default_dir: '{{ zk_tarball_installation | ternary("C:/tools/zookeeper-" + zk_version, "C:/tools/zookeeper-3.4.9") }}'
```
Comment (ViachaslauKabak) [inline vars/Win32NT.yml]: could we use path like for solr?
c:\zookeeper\zookeeper-version

File: vars/Win32NT.yml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+---
+zk_install_type: '{{ zk_tarball_installation | ternary("tarball_install", "choco_install") }}'
```
Comment (ViachaslauKabak) [inline vars/Win32NT.yml]: Need to add info about this to readme

File: templates/myid.j2 [hunk 14]
```diff
@@ -1,6 +1,6 @@
 {%- set ips = [] %}
 {%- for host in groups[zk_inventory_group] %}
-{{- ips.append(dict(id=loop.index, host=host, ip=hostvars[host]['ansible_default_ipv4'].address)) }}
+{{- ips.append(dict(id=loop.index, host=host, ip=hostvars[host]['ansible_fqdn'])) }}
```
[the remark this answers is not in the document]
Comment (koushyk) [reply templates/myid.j2]: windows facts does not have ansible_default_ipv4, they have only ansible_ip_asddresses it shows all IP address v6 v4.

File: vars/Win32NT.yml (hunk 13 above)
Comment (pavelpikta) [reply vars/Win32NT.yml]: resolve in fdd2c31

**Comment by pavelpikta** [comment]
TODO: on windows
* fix download with ckecksum
* fix zk_force_myid: false
* fix zk_force_config: false

**Pull request closed without merging**